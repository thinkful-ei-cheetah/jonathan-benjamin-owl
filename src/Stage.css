.container {
  max-width: 800px;
  margin: 0 auto;
  border: 1px solid black; 
}

.grid-row {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
}

.grid-item {
  width: 200px;
  padding: 10px;
}

.col-1 {
  flex-basis: 33.33%;
}

.col-2 {
  flex-basis: 66%;
}

.col-3 {
  flex-basis: 100%;
}