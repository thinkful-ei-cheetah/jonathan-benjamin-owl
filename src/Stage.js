import React, { Component } from 'react';
import './Stage.css';

class Stage extends Component {
  render() {
    return <div className="container grid-row">
        {this.props.onStage.map((person) =>
            <div className="grid-item col-1"> 
                <div className="person-name">{person.name}</div>   
                <img className="avatar-on-stage" src={person.avatar} alt="{person.name} avatar" />    
            </div>
        )}
    </div>
  }
}

export default Stage;