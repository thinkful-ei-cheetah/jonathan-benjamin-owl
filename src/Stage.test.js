import React from 'react';
import ReactDOM from 'react-dom';
import Stage from './Stage.js';
import renderer from 'react-test-renderer';

const participants = [
    {
        id: 1,
        name: 'Koren Templeton',
        avatar:
            'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
    }
]

//this is the test case
it('renders without crashing', () => {
    // first create a DOM element to render the component into
    const div = document.createElement('div');

    //render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(<Stage onStage={participants} />, div);

    //clean up code
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
        .create(<Stage onStage={participants}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
