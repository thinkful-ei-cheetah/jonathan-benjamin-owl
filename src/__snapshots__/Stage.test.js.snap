// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="container grid-row"
>
  <div
    className="grid-item col-1"
  >
    <div
      className="person-name"
    >
      Koren Templeton
    </div>
    <img
      alt="{person.name} avatar"
      className="avatar-on-stage"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
</div>
`;
